packages/grid/data-grid/src/tests/layout.DataGrid.test.tsx

```diff
@@ -32,6 +32,19 @@ describe('<DataGrid /> - Layout & Warnings', () => {
     columns: [{ field: 'brand' }],
   };
 
+  it('should throw an error if rows props is being mutated', () => {
+    expect(() => {
+      // We don't want to freeze baselineProps.rows
+      const rows = [...baselineProps.rows];
```

Comment:
it makes me want to add an options prop `disableFreezeRowsInDev`, what do you think? 🤔

Reply:
I don't understand the problem this prop would solve. The line looks expected. `baselineProps` is shared between the tests, we can't mutate it.

Reply:
Just an idea that came up when I saw that. Maybe users would want to share rows in tests or somehow opt out...
I think we can wait for a new issue if it comes up.

Reply:
I'd wait for a new issue to tackle it later.